### PR TITLE
stylish: init at unstable-2022-12-05

### PR DESCRIPTION
**Pull request by t-chab**
###### Description of changes

Add a package for [styli.sh](https://github.com/thevinter/styli.sh)

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -13718,6 +13718,12 @@
     githubId = 66133083;
     name = "Tomas Bravo";
   };
+  tchab = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:13721]: The maintainers list deserves its own commit.

No need for a PR, just split this in a new commit with the title `maintainers: add tchab`.

Tutorial: https://stackoverflow.com/questions/6217156/break-a-previous-commit-into-multiple-commits

File: pkgs/applications/misc/stylish/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "stylish";
+  version = "0.0.1";
+
+  src = fetchFromGitHub {
+    owner = "thevinter";
+    repo = "styli.sh";
+    rev = "d595412a949c6cdc7e151ae0cf929aa1958aa7f1";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/stylish/default.nix:13]: They do not use Git tags? It should be something like `rev = version;`

File: pkgs/applications/misc/stylish/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  installPhase = ''
+    mkdir -p $out/bin
+    cp "${src}/styli.sh" $out/bin
+    chmod +x $out/bin/styli.sh
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/thevinter/styli.sh";
+    description = "A shell script to manage wallpapers";
+    longDescription = ''
```
Comment (AndersonTorres) [inline pkgs/applications/misc/stylish/default.nix:26]: Break this in multiple lines, 80-char limit.

File: pkgs/applications/misc/stylish/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, stdenv
```
Comment (AndersonTorres) [inline pkgs/applications/misc/stylish/default.nix:2]: If your code does not use a C/C++ compiler, just use `stdenvNoCC`.

File: pkgs/applications/misc/stylish/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = fetchFromGitHub {
+    owner = "thevinter";
+    repo = "styli.sh";
+    rev = "d595412a949c6cdc7e151ae0cf929aa1958aa7f1";
+    hash = "sha256-lFnzrYnTFWe8bvK//aC1+TapWIFNsNP60Msn7D0tk/0=";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
```
Comment (AndersonTorres) [inline pkgs/applications/misc/stylish/default.nix:18]:
```suggestion
    runHook preInstall
    mkdir -p $out/bin
```

File: pkgs/applications/misc/stylish/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    cp "${src}/styli.sh" $out/bin
+    chmod +x $out/bin/styli.sh
```
Comment (AndersonTorres) [inline pkgs/applications/misc/stylish/default.nix:20]:
```suggestion
    chmod +x $out/bin/styli.sh
    runHook postInstall
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/stylish/default.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
+  pname = "stylish";
+  version = "0.0.1";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/stylish/default.nix:8]: If the upstream has no tags

https://github.com/thevinter/styli.sh/tags

use `version = unstable-YYYY-MM-DD;` where the date is the date of the commit you are using at the src.rev below.

File: pkgs/applications/misc/stylish/default.nix (hunk 2 above)
Comment (t-chab) [reply pkgs/applications/misc/stylish/default.nix:13]: I did not find any tags [on the upstream repository](https://github.com/thevinter/styli.sh/tags)

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (t-chab) [reply maintainers/maintainer-list.nix:13721]: Should be ok now.

File: pkgs/applications/misc/stylish/default.nix (hunk 3 above)
Comment (t-chab) [reply pkgs/applications/misc/stylish/default.nix:28]: Done.

File: pkgs/applications/misc/stylish/default.nix (hunk 4 above)
Comment (t-chab) [reply pkgs/applications/misc/stylish/default.nix]: Done

File: pkgs/applications/misc/stylish/default.nix (hunk 7 above)
Comment (t-chab) [reply pkgs/applications/misc/stylish/default.nix]: Version edited as requested.

File: pkgs/applications/misc/stylish/default.nix [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/thevinter/styli.sh";
+    description = "A shell script to manage wallpapers";
+    longDescription = ''
+      Styli.sh is a Bash script that aims to automate the tedious process
+of finding new wallpapers, downloading and switching them via the configs.
+Styli.sh can search for specific wallpapers from unsplash or download a random
+image from the specified subreddits. If you have pywal it also can set
+automatically your terminal colors.
```
Comment (AndersonTorres) [inline pkgs/applications/misc/stylish/default.nix:33]:
```suggestion
    longDescription = ''
      Styli.sh is a Bash script that aims to automate the tedious process
      of finding new wallpapers, downloading and switching them via the 
      configs.
      Styli.sh can search for specific wallpapers from unsplash or download
      a random image from the specified subreddits. If you have pywal it also
      can set automatically your terminal colors.
```

File: pkgs/applications/misc/stylish/default.nix [hunk 9]
```diff
@@ -0,0 +1,38 @@
+{ lib
```
Comment (AndersonTorres) [inline pkgs/applications/misc/stylish/default.nix:1]: Change the title of this commit:

`stylish: init at unstable-2022-12-05`

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/stylish/default.nix (hunk 9 above)
Comment (t-chab) [reply pkgs/applications/misc/stylish/default.nix:1]: Fixed.
